src/app/(proper_react)/redesign/(authenticated)/user/(dashboard)/settings/View.tsx

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Session } from "next-auth";
+import styles from "./View.module.scss";
+import { Toolbar } from "../../../../../../components/client/toolbar/Toolbar";
+import { ExtendedReactLocalization } from "../../../../../../hooks/l10n";
+import { EmailRow } from "../../../../../../../db/tables/emailAddresses";
+import { OpenInNew } from "../../../../../../components/server/Icons";
+import { EmailListing } from "./EmailListing";
+import { EmailAddressAdder } from "./EmailAddressAdder";
+import { AlertAddressForm } from "./AlertAddressForm";
+
+export type Props = {
+  l10n: ExtendedReactLocalization;
+  user: Session["user"];
+  monthlySubscriptionUrl: string;
+  yearlySubscriptionUrl: string;
+  fxaSettingsUrl: string;
+  emailAddresses: EmailRow[];
+  breachCountByEmailAddress: Record<string, number>;
+};
+
+export const SettingsView = (props: Props) => {
+  const l10n = props.l10n;
+
+  return (
+    <div className={styles.wrapper}>
+      <Toolbar
+        user={props.user}
+        monthlySubscriptionUrl={props.monthlySubscriptionUrl}
+        yearlySubscriptionUrl={props.yearlySubscriptionUrl}
+        fxaSettingsUrl={props.fxaSettingsUrl}
+      />
+      <main>
+        <header className={styles.title}>
+          <h2>{l10n.getString("settings-page-title")}</h2>
+        </header>
+        <div className={styles.content}>
+          <div>
+            <h3>{l10n.getString("settings-email-list-title")}</h3>
+            <p className={styles.description}>
+              {l10n.getString("settings-email-limit-info", {
+                limit: process.env.NEXT_PUBLIC_MAX_NUM_ADDRESSES!,
+              })}
+            </p>
+          </div>
+          <ul className={styles.emailList}>
+            <li key="primary">
+              <EmailListing
+                email={props.user.email}
+                breachCount={props.breachCountByEmailAddress[props.user.email]}
+              />
+            </li>
+            {props.emailAddresses.map((emailAddress) => {
+              return (
+                <li key={emailAddress.email}>
+                  <EmailListing
+                    email={emailAddress}
+                    breachCount={
+                      props.breachCountByEmailAddress[emailAddress.email]
+                    }
+                  />
+                </li>
+              );
+            })}
+          </ul>
+          <EmailAddressAdder />
+          <hr />
+          <AlertAddressForm
+            defaultSelected={
+              props.user.subscriber?.all_emails_to_primary
+                ? "primary"
+                : "affected"
+            }
```

Comment:
Not sure if this is only an issue with my local setup: After reloading the page the `primary` email option is always selected by default. Looking into my local database `all_email_to_primary` always is set to `false`.

Reply:
Oh that's a very good catch. So besides setting the wrong value (fixed in c271c1623), I also read the existing pref from the session token - which didn't get refreshed after the save. Fixed in dbb73f5f1.